Add all/individual setters for StyleBox default margins and StyleBoxTexture margin size and unbind `StyleBox*.set_*_individual()` methods

fire-forge opened the pull request:
StyleBoxFlat has methods for setting border width, corner radius, and expand margin for all sides or for each individual side. This PR adds similar methods for StyleBox default margins and StyleBoxTexture margin sizes.

This also unbinds the `StyleBox*.set_*_individual()` methods to simplify the API. These methods can still be used internally, and are used frequently for editor theme and default theme generation.

#### Additions:
* StyleBox:
    * `set_default_margin_all(float offset)`
    * `set_default_margin_individual(float offset_left, float offset_top, float offset_right, float offset_bottom)`
* StyleBoxTexture:
    * `set_margin_size_all(float size)`
    * `set_margin_size_individual(float size_left, float size_top, float size_right, float size_bottom)`

#### Unbinds:
- `StyleBox.set_default_margin_individual()`
- `StyleBoxFlat.set_corner_radius_individual()`
- `StyleBoxFlat.set_expand_margin_individual()`
- `StyleBoxTexture.set_expand_margin_individual()`
- `StyleBoxTexture.set_margin_size_individual()`

fire-forge committed:
Add set_default_margin_all/individual to StyleBox

YuriSizov reviewed (approved):
I approve the idea, though this needs a rebase. Not sure if the naming scheme is ideal, though. It's a bit hard to appreciate the difference between `set_margin`, `set_margin_all`, and `set_margin_individual`.

A web developer in me would probably accept `set_margin_all` that accepts 1, 2 or 4 arguments, and depending on the amount it sets all of them/pairs by axis/individuals. But that would likely be very unconventional for Godot API.

So I don't have a better suggestion. We do use these variants quite a lot, though, so addition definitely makes sense. We could avoid exposing `*_individual`, I guess, since user code will unlikely be as verbose as our generated themes.

fire-forge commented:
@YuriSizov I don't think it's a bad idea to unexpose the `*_individual` methods. That would clean up the API quite a bit. Worst case, if someone needs to set each side individually, they just write a function for it.

fire-forge committed:
Unbind StyleBox*.set_*_individual() methods
Unbinds the following methods:
- StyleBox.set_default_margin_individual()
- StyleBoxFlat.set_corner_radius_individual()
- StyleBoxFlat.set_expand_margin_individual()
- StyleBoxTexture.set_expand_margin_individual()
- StyleBoxTexture.set_margin_size_individual()

fire-forge commented:
@YuriSizov I rebased and split this PR into 2 commits: the first adds the new methods for StyleBox default margins and StyleBoxTexture margin size, and the second unbinds all of the `*_individual()` methods.

YuriSizov commented:
Oh, wait, we already have a variety of `set_*_individual` methods for other properties? I didn't realize that 😅 Don't use the API of `StyleBox` in projects much. In that case my suggestion may not be a good one...

fire-forge commented:
@YuriSizov Personally I'm indifferent as to whether the methods should be bound or not, so I think a second opinion would be good too. I put it as 2 separate commits so the unbinding can be reverted easily.

YuriSizov commented:
I'm not getting much feedback, so let's go with my gut feeling I guess. Let's unexpose `*_individual` methods, and reconsider that in future if someone comes forward missing them.

@fire-forge Needs a rebase, then should be good to merge.

fire-forge commented:
@YuriSizov Rebased!

akien-mga commented:
`<argument` should be renamed to `<param` in the docs: https://github.com/godotengine/godot/runs/8216431105?check_suite_focus=true

akien-mga commented:
Thanks!
